### PR TITLE
Release/3.0.0

### PR DESCRIPTION
**Pull request by DZariusz**
### Added
- New `OnChainDataFetcher` fetcher definition
- Add support of new numeric type: `FIXED.`

### Changed
- restrict packages versions with `~`

### CODE REVIEW

File: src/config/feeds-schema.ts [hunk 1]
```diff
@@ -205,5 +206,24 @@ export default {
       required: ['params'],
       additionalProperties: false,
     },
+    OnChainDataFetcher: {
+      properties: {
+        name: { const: 'OnChainData' },
+        params: {
+          type: 'object',
+          properties: {
+            address: { type: 'string' },
```
Comment (avoloshko) [inline src/config/feeds-schema.ts:215]: do you think we can use contract name through ENS?

**Comment by avoloshko** [comment]
Can we wait for pegasus PR before merging this?

File: src/config/feeds-schema.ts (hunk 1 above)
Comment (DZariusz) [reply src/config/feeds-schema.ts:215]: yes ENS can be used but this is not good idea. we should use addresses always.

**Comment by DZariusz** [comment]
> Can we wait for pegasus PR before merging this?

yes, we should merge after everything else is ready. btw all the PRs are open.

File: src/services/LeafValueCoder.ts [hunk 2]
```diff
@@ -22,6 +33,20 @@ export class LeafValueCoder {
...
    * @param leafAsHex {string} data in hex format
+   * @param label {string}
    */
-  static decode = (leafAsHex: string): number => new BigNumber(leafAsHex, 16).div(NUMERIC_MULTIPLIER).toNumber();
+  static decode = (leafAsHex: string, label: string): number | string => {
+    const bn = new BigNumber(leafAsHex, 16);
+    return LeafValueCoder.isFixedValue(label) ? bn.toFixed() : bn.div(NUMERIC_MULTIPLIER).toNumber();
+  };
+
+  static isFixedValue = (label: string): boolean => label.startsWith(FIXED_NUMBER_PREFIX);
```
Comment (avoloshko) [inline src/services/LeafValueCoder.ts:43]: how does this work? we're using keys to detect data type now?

File: test/services/LeafValueCoder.test.ts [hunk 3]
```diff
@@ -24,11 +25,9 @@ describe('LeafValueCoder', () => {
         [1.900719925474099, '1a60b6b3a763ff38'],
       ];
 
-      dataToTest.forEach((arr) => {
-        const [toConvert, expected] = arr;
-
+      dataToTest.forEach(([toConvert, expected]) => {
         it(`expect to convert ${toConvert} to solidity uint256`, () => {
-          const encoded = LeafValueCoder.encode(toConvert).toString('hex');
+          const encoded = LeafValueCoder.encode(toConvert, '').toString('hex');
```
Comment (avoloshko) [inline test/services/LeafValueCoder.test.ts:30]: what `''` means as the second parameter?

File: src/services/LeafValueCoder.ts (hunk 2 above)
Comment (DZariusz) [reply src/services/LeafValueCoder.ts:43]: yes

File: test/services/LeafValueCoder.test.ts (hunk 3 above)
Comment (DZariusz) [reply test/services/LeafValueCoder.test.ts:30]: empty/any label

**Comment by DZariusz** [comment]
Im merging this because we live and we need this v3

**Pull request merged**